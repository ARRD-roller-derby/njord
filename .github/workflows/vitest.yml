name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30