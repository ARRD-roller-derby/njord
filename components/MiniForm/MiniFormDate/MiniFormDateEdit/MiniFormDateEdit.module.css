.input {
  width: 100%;
}

.input > div {
  width: 100%;
}

.input input {
  width: 100%;
}