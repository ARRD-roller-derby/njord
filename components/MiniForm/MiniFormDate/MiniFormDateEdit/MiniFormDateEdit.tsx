import MiniFormSelectView from './MiniFormDateEditView'
import dayjs from 'dayjs';

interface props {
  readonly value?: string
  readonly setValue?: Function
}

export default function MiniFormDateEdit({value, setValue}: props) {
  return (
    <MiniFormSelectView
      setValue={setValue}
      value={dayjs(value).toDate()}
    />
  )
}
