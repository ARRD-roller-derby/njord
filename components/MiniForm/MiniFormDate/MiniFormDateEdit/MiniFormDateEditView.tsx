import classes from './MiniFormDateEdit.module.css'
import DatePicker from "react-datepicker";
interface props {
  readonly value: Date
  readonly setValue: Function
}

export default function MiniFormDateEditView({
  value,
  setValue,
}: props) {
  return (
    <div className={classes.input}>
      <DatePicker
        selected={value}
        showYearDropdown
        locale="fr"
        dateFormat="dd/MM/yyyy"
        yearDropdownItemNumber={50}
        scrollableYearDropdown
        onChange={(select)=>setValue(select)}
        popperClassName="datepicker"
      />
    </div>
  )
}
