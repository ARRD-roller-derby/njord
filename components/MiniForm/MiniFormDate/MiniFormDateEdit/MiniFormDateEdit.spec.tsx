import { afterEach, describe, expect, it, vi } from 'vitest'
import { render, cleanup, renderHook, act } from '@testing-library/react'
import useMiniFormDateEdit from './useMiniFormDateEdit';
import MiniFormDateEdit from './MiniFormDateEdit';
import dayjs from 'dayjs';

describe('<MiniFormDate />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(
      <MiniFormDateEdit setValue={vi.fn} value='2021-12-31' />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Test hook functions', () => {
    const 
      setValueMock=vi.fn(),
      { result } = renderHook(() => useMiniFormDateEdit(dayjs('01/01/2022').toDate(),setValueMock))

    act(() => {
      result.current.setAllState({label:'date State',value:'02'})
    })
    expect(setValueMock).toBeCalledWith('02')
    
  })
})
