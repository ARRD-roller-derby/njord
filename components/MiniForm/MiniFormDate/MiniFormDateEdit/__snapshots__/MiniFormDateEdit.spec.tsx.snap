// Vitest Snapshot v1

exports[`<MiniFormDate /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_input_1l1qv_1"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          type="text"
          value="31/12/2021"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
